LEX.CC missing <algorithm> include

## Rationale

when compiling HIT from RAVEN the following error is raised:

```
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.15.26726\bin\HostX64\x64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -IC:\msys64\home\talbpw\.conda\envs\raven_libraries\include -IC:\msys64\home\talbpw\.conda\envs\raven_libraries\PC /Tplex.cc /Fobuild\temp.win-amd64-2.7\Release\lex.obj -std=c++11
cl : Command line warning D9002 : ignoring unknown option '-std=c++11'

lex.cc

C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.15.26726\include\xlocale(319): warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc

lex.cc(197): error C2039: 'max': is not a member of 'std'

C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.15.26726\include\vector(20): note: see declaration of 'std'

lex.cc(197): error C3861: 'max': identifier not found

lex.cc(267): warning C4267: 'return': conversion from 'size_t' to 'int', possible loss of data

error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\VC\\Tools\\MSVC\\14.15.26726\\bin\\HostX64\\x64\\cl.exe' failed with exit status 2
```


## Description
```include <algorithm>``` needs to be added in lex.cc

## Impact

minimal

--- a/framework/contrib/hit/lex.cc
+++ b/framework/contrib/hit/lex.cc
@@ -1,6 +1,7 @@
 
 #include <string>
 #include <vector>
+#include <algorithm>
 
 #include "lex.h"
 

--- a/framework/contrib/hit/lex.cc
+++ b/framework/contrib/hit/lex.cc
@@ -1,7 +1,6 @@
 
 #include <string>
 #include <vector>
-
 #include <algorithm>
 
 #include "lex.h"

--- a/framework/contrib/hit/lex.cc
+++ b/framework/contrib/hit/lex.cc
@@ -1,6 +1,7 @@
 
 #include <string>
 #include <vector>
+
 #include <algorithm>
 
 #include "lex.h"

